controller/src/bin/xcvradm.rs

```diff
@@ -632,6 +836,109 @@ fn load_known_write_data<R: Read>(rdr: &mut R, kind: InputKind) -> anyhow::Resul
     }
 }
 
+#[derive(Clone, ValueEnum, PartialEq)]
+enum StatusFields {
+    /// The port number of the switch.
+    Port,
+    /// The power state of the transceiver module.
+    Status,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum PowerFields {
+    /// The port number of the switch.
+    Port,
+    /// The power state of the transceiver module.
+    Power,
+    /// Flag to indicate a software override of the modules power state.
+    Override,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum IdentifyFields {
+    /// The port number of the switch.
+    Port,
+    /// SFF-8024 identifier of the transceiver module.
+    Ident,
+    /// ??? - Needs details
+    Description,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum VendorInfoFields {
+    /// The port number of the switch.
+    Port,
+    /// Identifier of the transceiver module.
+    Identifier,
+    /// Manufacturer of the transceiver module.
+    Vendor,
+    /// Manufacturer part number.
+    Part,
+    /// Revision number of the transceiver module.
+    Rev,
+    /// Serial number of the transceiver module.
+    Serial,
+    /// ??? - Needs details
```

Comment:
This is a date-code specified in SFF-8636. It consists of:

- 2 low-order digits of the year ("22" in "2022")
- 2 digits for the month (Jan == "01")
- 2 digits for the day of the month
- 2 digits for a vendor-specific lot code, which could be blank.